OSX: clang compilation error: `use of undeclared identifier 'PATH_MAX'`

## Description

clang compilation error on OSX: `use of undeclared identifier 'PATH_MAX'`

## Steps to Reproduce

- Using a mac (tested on OSX High Sierra 10.12.2), follow the installation instructions in README. 
- The error happens when running `make install` in the `build` directory

```
/Users/maxwellheiber/dev/comdb2/berkdb/mp/mp_bh.c:578:11: error: use of undeclared identifier 'PATH_MAX'
        char dir[PATH_MAX];
                 ^
1 error generated.
make[2]: *** [berkdb/CMakeFiles/db.dir/mp/mp_bh.c.o] Error 1
make[1]: *** [berkdb/CMakeFiles/db.dir/all] Error 2
make: *** [all] Error 2
```

Versions:

```
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.3.0
```

comdb2 compiles and runs if a definition for `PATH_MAX` is added to the top of the file.

--- a/berkdb/mp/mp_bh.c
+++ b/berkdb/mp/mp_bh.c
@@ -13,6 +13,7 @@ static const char revid[] = "$Id: mp_bh.c,v 11.86 2003/07/02 20:02:37 mjc Exp $"
 
 #ifndef NO_SYSTEM_INCLUDES
 #include <alloca.h>
+#include <limits.h>
 #include <sys/types.h>
 #include <stdlib.h>
 #include <unistd.h>

